specification/quantumleap.yml

```diff
@@ -0,0 +1,500 @@
+        400:
+          description: "Received notification is not valid."
+        500:
+          description: "Internal server error."
+
+################################################################################
+# PATHS: OUTPUT
+################################################################################
+  /entities/{entityId}/attrs/{attrName}:
+    get:
+      operationId: reporter.reporter.query_1T1E1A
+      summary: "History of an attribute of a given entity instance."
+      description: "For example, query max water level of the central tank throughout the last year. Queries can get more sophisticated with the use of filters and query attributes."
+      tags:
+        - queries
+      parameters:
+        - $ref: '#/parameters/entityIdInPath'
+        - $ref: '#/parameters/attrNameInPath'
+        # Query attrs to be refined...
+        - $ref: '#/parameters/type'
+        - $ref: '#/parameters/aggrMethod'
+        - $ref: '#/parameters/aggrPeriod'
+        - $ref: '#/parameters/options'
+        - $ref: '#/parameters/from'
+        - $ref: '#/parameters/to'
+        - $ref: '#/parameters/lastN'
+        - $ref: '#/parameters/limit'
+      responses:
+        200:
+          description: OK
+          # Response types to be refined...
+          schema:
+            type: object
+            properties:
+              entityId:
+                type: string
+              attrName:
+                type: string
+              index:
+                type: array
+                items:
+                  type: number
+                description: "Array of the timestamps of the requested data. Values are UTC milliseconds since 1970-01-01T00:00:00Z."
+              values:
+                type: array
+                items:
+                  type: number
+                description: "Array of values of the selected attribute, in the same corresponding order of the 'index' array."
+          examples:
+            application/json:
+              {
+                "entityId": "Kitchen1",
+                "attrName": "temperature",
+                "index": [1510826981388, 1510826990438, 1510826999725],
+                "values": [24.1, 25.3, 26.7]
+              }
+
+  # TODO: Could we remove all these /value suffixed endpoints if we provide the 'options' query arg? (keyvalues or values)?
+  /entities/{entityId}/attrs/{attrName}/value:
+    get:
+      operationId: reporter.reporter.query_1T1E1A_value
+      summary: "History of an attribute (values only) of a given entity instance."
+      description: "Similar to the previous, but focusing on the values regardles of the metadata."
+      tags:
+        - queries
+      parameters:
+        - $ref: '#/parameters/entityIdInPath'
+        - $ref: '#/parameters/attrNameInPath'
+        # Query attrs to be refined...
+        - $ref: '#/parameters/type'
+      responses:
+        200:
+          description: OK
+          # Response types to be refined...
```

Comment:
```
{
  "index": [1510826981388, 1510826990438, 1510826999725],
  "values": [24.1, 25.3, 26.7]
}
```

Reply:
This brings us to an interesting point to decide, the UTC in milliseconds vs ISO8601.
Technically the UTC in milliseconds is more "ideal", but looking at the context, if the user/notification is already bringing data in in ISO8601, then I believe the response should be consistent and reply the same format. If course, if the user inserted UTC milliseconds, the response will be like your example above. 
What do you think?

Reply:
probably it could be a parameter, e.g. indexFormat and possible values are UTC and ISO8601 where ISO8601 is the default

Reply:
btw how the index are stored in crate?